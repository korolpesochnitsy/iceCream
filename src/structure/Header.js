import React from 'react';
import FocusLink from '../structure/FocusLink';
import iceCream from '../assets/img/ultimate-ice-cream.svg';
import { css } from 'emotion/macro';

const headerStyle = css`
  position: relative;
  text-align: center;
  padding-top: 3em;

  h1 {
    display: flex;
    justify-content: center;
    color: #313030;
    font-weight: bold;
    font-family: 'cornerstone', sans-serif;
    font-size: 2.5em;

    img {
      margin-right: 0.5em;
    }
  }

  nav {
    max-width: 63.75em;
    margin-left: auto;
    margin-right: auto;
    margin-top: 3em;

    padding: 0.5em;
    background-color: #ffffff;
    border-radius: 7em;
    border: 1px solid rgba(32, 33, 36, 0.12);
    background-clip: padding-box;
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.05);

    display: flex;
    align-items: center;

    a {
      position: relative;
      color: #5c4268;
      border: 2px solid transparent;
      border-radius: 6em;
      padding: 0 0.75em;
      font-size: 1em;
      line-height: 2em;
      text-decoration: none;
      transition: box-shadow 0.2s ease-in-out;

      &:hover {
        text-decoration: underline;
      }

      &:nth-of-type(n + 2) {
        &:before {
          content: '';
          position: absolute;
          left: -2px;
          top: 15%;
          height: 70%;
          width: 1px;
          background: rgba(32, 33, 36, 0.1);
        }
      }

      &.active {
        color: #a84a7a;
      }

      &:focus:not(:active) {
        outline: 2px solid transparent;
        box-shadow: 0 0 0 2px #8b9099;

        &.active {
          box-shadow: 0 0 0 2px #a84a7a;
        }
      }
    }
  }
`;

const Header = () => (
  <header className={headerStyle}>
    <h1>
      <img src={iceCream} alt="" />
      Ice Cream
    </h1>
    <nav>
      <FocusLink to="/" activeClassName="active" exact>
        Menu
      </FocusLink>
      <FocusLink to="/ice-creams" activeClassName="active">
        Add Ice Cream
      </FocusLink>
    </nav>
  </header>
);

export default Header;
