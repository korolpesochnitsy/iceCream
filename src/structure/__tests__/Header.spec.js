import React from 'react';
import { render } from '@testing-library/react';
import Header from '../Header';

describe('Header', () => {
  it('should render', () => {
    const { container } = render(<Header />);
    const img = container.firstChild.querySelector('header > h1 > img');
    expect(img).toHaveAttribute('alt', '');
    expect(img).toHaveAttribute('src', 'ultimate-ice-cream.svg');

    const allAnchors = container.firstChild.querySelectorAll('nav a');
    expect(allAnchors[0]).toHaveAttribute('href', '/');
    expect(allAnchors[0]).toHaveTextContent('Menu');
    expect(allAnchors[1]).toHaveAttribute('href', '/ice-creams');
    expect(allAnchors[1]).toHaveTextContent('Add Ice Cream');

    const navContainer = container.firstChild.querySelector('nav');
    expect(navContainer).toMatchSnapshot();
  });
});
