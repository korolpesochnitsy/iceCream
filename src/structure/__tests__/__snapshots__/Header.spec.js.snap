// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<nav>
  <a
    href="/"
  >
    Menu
    <span>
      {"focus":true}
    </span>
    <span
      data-testid="linkType"
    >
      NavLink
    </span>
  </a>
  <a
    href="/ice-creams"
  >
    Add Ice Cream
    <span>
      {"focus":true}
    </span>
    <span
      data-testid="linkType"
    >
      NavLink
    </span>
  </a>
</nav>
`;
